### PR TITLE
Updated Apps, payment packages

### LINKED ISSUE #1408: Group payment methods into single app card
Move all payment methods to a single  "Payments" app card.   

All payment methods settings to go into single admin settings view, ala Taxes. 

Related to #1383

### PR DESCRIPTION
**Pull request by aaronjudd**
- single card payment layout
- updated / refactored Reaction.Apps
- normalization of client Packages moved to publication
- PayPal and PayFlow split out
- removes default payment method and collapse handling in checkout
- commented dashboard default method settings. TBD.
- resolves #1408
- adds object-curly-spacing lint rule
- adds significant i18n to payments, makes local
- adds translations to month helpers

### CODE REVIEW

File: imports/plugins/core/layout/client/templates/layout/admin/admin.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import Drop from "tether-drop";
-import { Meteor } from "meteor/meteor";
-import { Blaze } from "meteor/blaze";
-import { Template } from "meteor/templating";
-import { Reaction, i18next } from "/client/api";
-import { Packages } from "/lib/collections";
-
+import {Meteor} from "meteor/meteor";
```
Comment (zenweasel) [inline imports/plugins/core/layout/client/templates/layout/admin/admin.js]: Shouldn't we keep the spaces around imports consistent?

**Comment by aaronjudd** [comment]
No, this should go ahead and be merged -> it does have the placeholders for rates, and codes, but it's not really related functionality at this point (although had to be done to add codes in the checkout the way I want)

File: imports/plugins/core/layout/client/templates/layout/admin/admin.js (hunk 1 above)
Comment (aaronjudd) [inline imports/plugins/core/layout/client/templates/layout/admin/admin.js]: Apparently a beautifier rule that wasn't working before.

File: imports/plugins/core/discounts/server/methods/methods.js [hunk 2]
```diff
@@ -53,29 +53,17 @@ export const methods = {
...
-      Meteor.call("discounts/setRate", cartToCalc._id, discountRate);
-    } else {
-      // we are here because the custom rate package is disabled.
-      // we're going to set an inital rate of 0
-      // all methods that trigger when discounts/calculate will
-      // recalculate this rate as needed.
-      // Meteor.call("discounts/setRate", cartToCalc._id, discountRate);
+    if (cartToCalc.discountMethods) {
+      for (discountMethods of cartToCalc.discountMethods) {
+        // get discount based on id in cartMethods.
```
Comment (zenweasel) [inline imports/plugins/core/discounts/server/methods/methods.js]: Yeah, just wanted to double-check that this intentionally does nothing here

File: client/modules/core/helpers/apps.js [hunk 3]
```diff
@@ -1,25 +1,15 @@
-import { Reaction, Logger } from "/client/api";
-import { Packages } from "/lib/collections";
-import { Template } from "meteor/templating";
+import _ from "lodash";
+import {Reaction} from "/client/api";
+import {Packages} from "/lib/collections";
+import {Template} from "meteor/templating";
```
Comment (jshimko) [inline client/modules/core/helpers/apps.js]: Bracket spacing, import order.

Comment (aaronjudd) [inline client/modules/core/helpers/apps.js]: do we have a rule to enforce this?     `"space_in_paren": true` in the _.jsbeautifyrc_ will add the spaces on save, but also adds in a bunch of other places.  Eslint doesn't see anything wrong here..  which would leave me back to disabling .jsbeautify.. (nobody else using this I guess?).

File: client/modules/i18n/startup.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { Reaction } from "/client/api";
 import { Shops, Translations } from "/lib/collections";
 import * as Schemas from "/lib/collections/schemas";
 import i18next, { packageNamespaces, getLabelsFor, getMessagesFor, i18nextDep } from "./main";
+import {mergeDeep} from "/lib/api";
```
Comment (jshimko) [inline client/modules/i18n/startup.js]: Looks like we need to add the `space-in-brackets` rule (and make arrays an exception).

http://eslint.org/docs/rules/space-in-brackets

Probably something like...

```js
"space-in-brackets": [2, "always", {
  "objectsInArrays": false,
  "arraysInArrays": false,
  "arraysInObjects": false
}]
```

Comment (aaronjudd) [inline client/modules/i18n/startup.js]: This rule was removed in ESLint v1.0 and replaced by the object-curly-spacing and array-bracket-spacing rules.

Comment (jshimko) [inline client/modules/i18n/startup.js]: Alrighty...

```js
"object-curly-spacing": [2, "always"],
```

Comment (aaronjudd) [inline client/modules/i18n/startup.js]:
```  
  "sort-imports": ["error", {
        "ignoreCase": false,
        "ignoreMemberSort": false,
        "memberSyntaxSortOrder": ["none", "all", "multiple", "single"]
    }],
    "object-curly-spacing": ["error", "always", { "objectsInObjects": true }],
```

However, that's going to change 90 files, as --fix rightfully adds spacing to objects all over the place.  I'm actually cool with that, but it's new... 

Sorts rule isn't quite advanced enough to enforce our rules (I think this basically is just alphabetical).

**Comment by zenweasel** [comment]
Checking out with the Example payment method I see this error in the console. Order does complete however.

```

meteor.js?hash=e3f53db…:930 Exception in template helper: TypeError: moment is not a function
    at Object.<anonymous> (http://localhost:3000/app/app.js?hash=65db4f0c5205ad7c0f0dd9cb714deba07c0401e4:50923:12)
    at http://localhost:3000/packages/blaze.js?hash=10a350f1c0d55cf44d3c99989b04ef2e7e0d23f6:3035:16
    at http://localhost:3000/packages/blaze.js?hash=10a350f1c0d55cf44d3c99989b04ef2e7e0d23f6:1697:16
    at http://localhost:3000/packages/blaze.js?hash=10a350f1c0d55cf44d3c99989b04ef2e7e0d23f6:3087:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=10a350f1c0d55cf44d3c99989b04ef2e7e0d23f6:3728:12)
    at http://localhost:3000/packages/blaze.js?hash=10a350f1c0d55cf44d3c99989b04ef2e7e0d23f6:3086:27
    at Spacebars.call (http://localhost:3000/packages/spacebars.js?hash=ebf9381e7fc625d41acb0df14995b7614360858a:172:18)
    at Spacebars.mustacheImpl (http://localhost:3000/packages/spacebars.js?hash=ebf9381e7fc625d41acb0df14995b7614360858a:106:25)
    at Object.Spacebars.mustache (http://localhost:3000/packages/spacebars.js?hash=ebf9381e7fc625d41acb0df14995b7614360858a:110:39)
    at Blaze.View._render (http://localhost:3000/app/app.js?hash=65db4f0c5205ad7c0f0dd9cb714deba07c0401e4:25685:24)
```

**Comment by zenweasel** [comment]
I wonder if we shouldn't throw up a warning in the settings panel if no Payment packages are enabled?

**Comment by zenweasel** [comment]
Seeing this message (not an error) in the console when PayPal PayFlow is enabled:

```
modules.js?hash=09d32e0…:47071 Deprecation warning: use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale See http://momentjs.com/guides/#/warnings/define-locale/ for more info.
```

**Comment by zenweasel** [comment]
It doesn't look like we do a client-side validation on Expiration date? For Paypal Payflow I got this error when I submitted a card with an invalid expiration.

cart_checkout

**Comment by aaronjudd** [comment]
I did update the way `moment` is loading, so the first error could be related (try to get to that one reproduced right now).  Re: the deprecation warning..  that's because we need more garbage debugging on that page that we can't shut off.  oh no wait, it's because I am using defineLocale to add an alias to ZH, which is what it seems to be indicating is the correct purpose, not the deprecated purpose.... I think..

**Comment by aaronjudd** [comment]
re: client side validation - no, there is not any validation to speak of other than autoform stuff.  nothing new here.  however, I've been talking about having a credit card / payment method component for a while, to replace this input form with a nice modern input validator, aka card.js etc.   I was holding myself back, but if you are suggesting it's a must have..

**Comment by zenweasel** [comment]
Approved - Except for the couple of minor things I think this is a nice step forward, especially on the consumer-side where I thought that part was probably confusing.

**Comment by zenweasel** [comment]
It seems like Paypal Express is enabled by default?

**Comment by zenweasel** [comment]
This checkbox looks a little odd (Paypal Express)

dashboard

**Comment by zenweasel** [comment]
If I have no payment methods specified, in FF, instead of seeing the "No Payment" message I see all payment methods

cart_checkout

**Comment by aaronjudd** [comment]
the way it's setup now, you'll see all the payment methods if you are logged in as admin in checkout, the view restriction is permission based now (or at least partially).  I'd like to see us narrow this down with an implementation of "audience" permissions checks.   The consumer/client should only be seeing what's enabled.

**Comment by aaronjudd** [comment]
re: defaults, I set the initial to have "PayPal express" and "Example" enabled.

**Comment by aaronjudd** [comment]
@zenweasel All issues you reported should now be addressed. Ready to merge.

**Pull request merged**